git/config.py

```diff
@@ -424,8 +425,11 @@ def read(self):
                     if include_path in seen or not os.access(include_path, os.R_OK):
                         continue
                     seen.add(include_path)
-                    files_to_read.append(include_path)
+                    include_paths.append(include_path)
                     num_read_include_files += 1
+                include_paths.reverse()
```

Comment:
I think this block should be rewritten like this:
```python
files_to_read = include_paths + files_to_read
```
This makes clearer that included files should be read and merged first.